### PR TITLE
ensure all github projects have correct template

### PR DESCRIPTION
**Pull request by goneri**
Ensure all the projects with either `noop-jobs` or `system-required`.

### CODE REVIEW

**Comment by pabelanger** [comment]
https://github.com/ansible/project-config/pull/285 was my attempt before to do this, but haven't continued with it.  I feel we should adapt this script, into existing tools/zuul-projects-checks.py.  I think the missing step, which you are doing, is loading the projects from zuul/tenants.yaml, over me hardcoding namespaces

File: tools/zuul-projects-checks.py [hunk 1]
```diff
@@ -170,6 +172,42 @@ def check_only_boilerplate():
     return errors
 
 
+def check_project_templates():
```
Comment (pabelanger) [inline tools/zuul-projects-checks.py:175]: we don't appear to call this function

File: zuul.d/projects.yaml [hunk 2]
```diff
@@ -10,6 +10,12 @@
     name: github.com/ansible-collection-migration/ansible-base
     default-branch: devel
 
+- project:
+    name: github.com/ansible-collections/amazon.aws
+    default-branch: master
+    templates:
```
Comment (pabelanger) [inline zuul.d/projects.yaml:16]: This isn't correct, we now want all projects to atleast have system-required. This is because we add the github-workflows jobs now.  Noop isn't needed, so we can skip this

File: tools/zuul-projects-checks.py [hunk 3]
```diff
@@ -170,6 +172,42 @@ def check_only_boilerplate():
...
+            elif github_project == name:
+                templates += project.get("templates", [])
+        if not templates:
+            print(
+                "No template for project {github_project}".format(
+                    github_project=github_project
+                )
+            )
+            errors = True
+        if "noop-jobs" not in templates:
```
Comment (pabelanger) [inline tools/zuul-projects-checks.py:199]: we no longer need to check for noop-jobs, just system-required.

File: tools/zuul-projects-checks.py (hunk 1 above)
Comment (goneri) [reply tools/zuul-projects-checks.py:175]: oh crap. Letś me fix that.

File: tools/zuul-projects-checks.py (hunk 3 above)
Comment (goneri) [reply tools/zuul-projects-checks.py:199]: Ok.

File: tools/zuul-projects-checks.py [hunk 4]
```diff
@@ -170,18 +173,54 @@ def check_only_boilerplate():
     return errors
 
 
+def check_project_templates():
```
Comment (pabelanger) [inline tools/zuul-projects-checks.py:176]: lets add something like other functions... eg:

```
print("\nChecking that every project has entries")
print("======================")
```

Then check errors, and output some instructions

File: zuul/tenants.yaml [hunk 5]
```diff
@@ -53,7 +53,6 @@
           - ansible-community/pytest-molecule
           - ansible/network
           - ansible-network/arista_eos
-          - ansible-network/aws
```
Comment (pabelanger) [inline zuul/tenants.yaml:56]: lets move this into another PR,  if no longer needed. As we need to retire via github/projects.yaml
